Unstructured-IO/pipeline-template
# feat: add logging config

Adds a proper `logger_config.yaml` for uvicorn.

**Test Instructions**
`make generate-repo` should prompt the following template fields:
```
pipeline_family [template]: test_family
repo_name [pipeline-test_family]: test_repo
shortname [test_family]: test
pipeline_package [test_family]: test_package
```
`cd` into `test_repo`.

Then, `make run-web-app` should include formatting output specific to `logger_config.yaml` as shown below.
```
(unstructured) ygyoon@Nathans-MacBook-Pro test_repo % make run-web-app
PYTHONPATH=/Users/ygyoon/pipeline-template/test_repo uvicorn prepline_test_package.api.app:app --log-config logger_config.yaml --log-config logger_config.yaml --reload
2023-04-05 13:38:49,188 uvicorn.error INFO Will watch for changes in these directories: ['/Users/ygyoon/pipeline-template/test_repo']
2023-04-05 13:38:49,188 uvicorn.error INFO Uvicorn running on http://127.0.0.1:8000 (Press CTRL+C to quit)
2023-04-05 13:38:49,188 uvicorn.error INFO Started reloader process [6600] using WatchFiles
2023-04-05 13:38:49,920 uvicorn.error INFO Started server process [6611]
2023-04-05 13:38:49,920 uvicorn.error INFO Waiting for application startup.
2023-04-05 13:38:49,920 uvicorn.error INFO Application startup complete.
```
Test hitting the API with:
```
curl -X 'POST' \
  'http://localhost:8000/test_family/v0.0.1/process-file' \
  -H 'accept: application/json' \
  -H 'Content-Type: multipart/form-data' \
  -F 'files=@test.txt' \   
  -F 'some_parameter=something'  | jq -C . | less -R
```
And note the output line from the `pipeline-test_family` logger:
```
...
2023-04-05 16:25:12,568 127.0.0.1:60537 POST /test_family/v0.0.1/process-file HTTP/1.1 - 200 OK
...
```
